### PR TITLE
fix: [PL-41926]: When listing delegate groups that are using the specified delegate token, check for the token validity

### PR DESCRIPTION
**Pull request by VikasMaddukuriHarness**
## Describe your changes
When listing delegate groups that are using the specified delegate token, we return an empty list even though the token does not exist. Post this change, we will throw an invalidRequest exception saying the token does not exist.

Screenshot 2023-10-24 at 1 33 30 PM

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54542)

### CODE REVIEW

**Comment by VikasMaddukuriHarness** [comment]
trigger uts

**Comment by VikasMaddukuriHarness** [comment]
trigger uts

File: 420-delegate-service/src/main/java/io/harness/service/impl/DelegateSetupServiceImpl.java [hunk 1]
```diff
@@ -394,6 +394,9 @@ private List<DelegateGroupDetails> getDelegateGroupDetails(final String accountI
                                         .in(delegateGroupIds);
 
     if (isNotEmpty(delegateTokenName)) {
+      if (!isDelegateTokenValid(accountId, delegateTokenName)) {
+        throw new InvalidRequestException("No such delegate token exists.");
```
Comment (xingchi-jin) [inline 420-delegate-service/src/main/java/io/harness/service/impl/DelegateSetupServiceImpl.java:398]: can we return NOT_FOUND ?

File: 420-delegate-service/src/main/java/io/harness/service/impl/DelegateSetupServiceImpl.java [hunk 2]
```diff
@@ -412,6 +415,16 @@ private List<DelegateGroupDetails> getDelegateGroupDetails(final String accountI
         .collect(toList());
   }
 
+  private boolean isDelegateTokenValid(String accountId, String tokenName) {
```
Comment (xingchi-jin) [inline 420-delegate-service/src/main/java/io/harness/service/impl/DelegateSetupServiceImpl.java:418]: i think we should move this check to listDelegateGroupDetails function. Check token existent before invoking getDelegateGroupDetails

Comment (xingchi-jin) [reply 420-delegate-service/src/main/java/io/harness/service/impl/DelegateSetupServiceImpl.java:418]: btw, there is DelegateNgTokenService class, which is a wrapper of the persistence. restriction is it only searches for token whose isNg flag is true. This restriction should be fine in this api

**Comment by VikasMaddukuriHarness** [comment]
trigger uts

**Pull request closed without merging**